lib/rules/sort-imports.js

```diff
@@ -0,0 +1,141 @@
+     * @param {ASTNode} node - the ImportDeclaration node.
+     * @returns {?string} the local name of the first imported module.
+     */
+    function getFirstLocalMemberName(node) {
+        if (node.specifiers[0]) {
+            return node.specifiers[0].local.name;
+        } else {
+            return null;
+        }
+    }
+
+    return {
+        "ImportDeclaration": function(node) {
+            if (previousDeclaration) {
+                var currentLocalMemberName = getFirstLocalMemberName(node),
+                    currentMemberSyntaxGroupIndex = getMemberParameterGroupIndex(node),
+                    previousLocalMemberName = getFirstLocalMemberName(previousDeclaration),
+                    previousMemberSyntaxGroupIndex = getMemberParameterGroupIndex(previousDeclaration);
+
+                if (ignoreCase) {
+                    previousLocalMemberName = previousLocalMemberName.toLowerCase();
+                    currentLocalMemberName = currentLocalMemberName.toLowerCase();
+                }
+
+                // When the current declaration uses a different member syntax,
+                // then check if the ordering is correct.
+                // Otherwise, make a default string compare (like rule sort-vars to be consistent) of the first used local member name.
+                if (currentMemberSyntaxGroupIndex !== previousMemberSyntaxGroupIndex) {
+                    if (currentMemberSyntaxGroupIndex < previousMemberSyntaxGroupIndex) {
+                        context.report(node, "Imports should be grouped by the member syntax. Use '{{syntaxA}}' before '{{syntaxB}}' member syntax.", {
+                            syntaxA: groupByMemberSyntax[currentMemberSyntaxGroupIndex],
+                            syntaxB: groupByMemberSyntax[previousMemberSyntaxGroupIndex]
+                        });
+                    }
+                } else {
+                    if (currentLocalMemberName < previousLocalMemberName) {
+                        context.report(node, "Imports should be sorted alphabetically.");
+                    }
+                }
+            }
+
+            // Multiple members of an import declaration should also be sorted alphabetically.
+            if (!ignoreMemberSort && node.specifiers.length > 1) {
+                node.specifiers.reduce(function(previousSpecifier, currentSpecifier) {
+                    var currentSpecifierName = currentSpecifier.local.name,
+                        previousSpecifierName = previousSpecifier.local.name;
+
+                    if (ignoreCase) {
+                        currentSpecifierName = currentSpecifierName.toLowerCase();
+                        previousSpecifierName = previousSpecifierName.toLowerCase();
+                    }
+
+                    if (currentSpecifierName < previousSpecifierName) {
+                        context.report(currentSpecifier, "Members of an import declaration should be sorted alphabetically.");
```

Comment:
Name of the member in the error message. Why I ask this is because in one of you invalid unit test, the rule throws 2 general messages for same import (which is kinda redundant).
I am asking whether this can be done or not?